apps/outlook-rooms-addin/src/tests/find-space.component.spec.ts

```diff
@@ -0,0 +1,260 @@
+                useClass: OrgServiceStub,
+            },
+            {
+                provide: SpacesService,
+                useClass: SpacesServiceStub,
+            },
+            {
+                provide: EventFormService,
+                useValue: {
+                    setView: jest.fn(() => {}),
+                    newForm: jest.fn(() => {}),
+                    clearForm: jest.fn(),
+                    storeForm: jest.fn(() => {}),
+                    loadForm: jest.fn(),
+                    postForm: jest.fn(),
+                    setOptions: jest.fn(() => {}),
+                    view: '',
+                    available_spaces: of(mockSpace, mockSpace, mockSpace),
+                    options: of(mockEventFlowOptions),
+                },
+            },
+            { provide: Location, useValue: RouterStub },
+            {
+                provide: FeaturesFilterService,
+                useClass: FeatureFilterServiceStub,
+            },
+            {
+                provide: MapService,
+                useClass: MapServiceStub,
+            },
+            { provide: RoomConfirmService, useClass: RoomConfirmServiceStub },
+            { provide: Router, useValue: { navigate: jest.fn() } },
+        ],
+        declarations: [
+            MockComponent(FindSpaceItemComponent),
+            MockComponent(InteractiveMapComponent),
+            MockComponent(FilterSpaceComponent),
+        ],
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        jest.resetModules();
+        ngMocks.reset();
+        spectator = createComponent();
+        spectator.inject(FeaturesFilterService);
+        const event_service: any = spectator.inject(EventFormService);
+        event_service.setOptions.mockImplementation(() => {
+            return of(mockEventFlowOptions);
+        });
+
+        event_service.setView.mockImplementation((_) => {
+            event_service.view = _;
+            spectator.detectChanges();
+        });
+        event_service.newForm.mockImplementation((_) => {
+            event_service.form = fb.group(formModel);
+        });
+        event_service.storeForm.mockImplementation((_) => {
+            if (
+                event_service.form.controls.title == null ||
+                event_service.form.controls.date == null ||
+                event_service.form.controls.duration == null
+            )
+                return null;
+        });
+    });
+
+    it('should create component', () => {
+        expect(spectator.component).toBeTruthy();
+    });
+    it('should open the Filter modal when clicked', () => {
+        const event_service = spectator.inject(EventFormService);
+        event_service.newForm();
+        const mat_bottom_sheet = spectator.inject(MatBottomSheet);
+        mat_bottom_sheet.open.mockImplementation((template, config) => {
+            return '' as any;
+        });
+        const open_filter_spy = jest.spyOn(mat_bottom_sheet, 'open');
```

Comment:
Functions already mocked out with `jest.fn()` don't need to be spied on as they are already spies.